packages/runtime-common/realm.ts

```diff
@@ -334,10 +336,19 @@ export class Realm {
     }
 
     // new instances are created in a folder named after the card
-    let dirName = `/${join(
-      new URL(this.url).pathname,
-      resource.meta.adoptsFrom.name
-    )}/`;
+    let name: string;
+    if (!("type" in resource.meta.adoptsFrom)) {
+      name = resource.meta.adoptsFrom.name;
+    } else {
+      let card = await loadCard(resource.meta.adoptsFrom);
+      if (!card) {
+        throw new Error(
+          `Card not found for ref ${JSON.stringify(resource.meta.adoptsFrom)}`
+        );
+      }
+      name = card.name;
```

Comment:
we used to have this function that could serialize the card ref into a meaningful name (it got factored out). let me hunt for it--it's probably be a good addition for the `card-ref.ts` module.

Reply:
it actually still exists, it's `internalKeyFor()` from `@cardstack/runtime-common/index.ts`.  however, it has some characteristics that probably don't suit this very well, as you'd need to trim off the leading module URL to use this.

Reply:
also, probably the card-ref module is probably a better home for `internalKeyFor()` (we can still reexport it thru the runtime-common index)

Reply:
`internalKeyFor` actually only gives the relative name for a card and not the card's actual name. Instead, I added a `name` field on `CardRef` interface for each card type, so now it's easier to get the name without loading the card